app/javascript/schools/founders/components/students_panel/SA_StudentsPanel_UpdateDetailsForm.re

```diff
@@ -0,0 +1,393 @@
+[@bs.config {jsx: 3}];
+
+open StudentsPanel__Types;
+
+type teamCoachlist = (string, string, bool);
+
+type state = {
+  name: string,
+  teamName: string,
+  tagsToApply: list(string),
+  teamCoaches: list(teamCoachlist),
+  excludedFromLeaderboard: bool,
+  title: string,
+  affiliation: string,
+  saving: bool,
+  accessEndsAt: option(Js.Date.t),
+};
+
+type action =
+  | UpdateName(string)
+  | UpdateTeamName(string)
+  | AddTag(string)
+  | RemoveTag(string)
+  | UpdateCoachesList(string, string, bool)
+  | UpdateExcludedFromLeaderboard(bool)
+  | UpdateTitle(string)
+  | UpdateAffiliation(string)
+  | UpdateSaving(bool)
+  | UpdateAccessEndsAt(option(Js.Date.t));
+
+let str = ReasonReact.string;
+
+let stringInputInvalid = s => s |> String.length < 2;
+
+let updateName = (send, name) => {
+  send(UpdateName(name));
+};
+
+let updateTeamName = (send, teamName) => {
+  send(UpdateTeamName(teamName));
+};
+
+let updateTitle = (send, title) => {
+  send(UpdateTitle(title));
+};
+
+let formInvalid = state =>
+  state.name
+  |> stringInputInvalid
+  || state.teamName
+  |> stringInputInvalid
+  || state.title
+  |> stringInputInvalid;
+
+let handleErrorCB = (send, ()) => send(UpdateSaving(false));
+
+let handleResponseCB = (submitCB, state, json) => {
+  let teams = json |> Json.Decode.(field("teams", list(Team.decode)));
+  let students =
+    json |> Json.Decode.(field("students", list(Student.decode)));
+
+  submitCB(teams, students, state.tagsToApply);
+  Notification.success("Success", "Student updated successfully");
```

Comment:
@bodhish: It will be a good idea to notify that the student/team is archived if the access ends at is set to an earlier date.